drizzlepac/hlautils/catalog_utils.py

```diff
@@ -131,57 +119,37 @@ def compute_background(self, box_size=BKG_BOX_SIZE, win_size=BKG_FILTER_SIZE,
+
             except Exception:
                 bkg = None
                 continue
 
             if bkg is not None:
-                # Set the bkg_rms at "nsigma" sigma above background
-                bkg_rms = nsigma * bkg.background_rms
-                default_threshold = bkg.background + bkg_rms
-                bkg_rms_mean = bkg.background.mean() + nsigma * bkg_rms.std()
-                bkg_mean = bkg.background.mean()
-                bkg_dao_rms = bkg.background_rms
-                if threshold_flag is None:
-                    threshold = default_threshold
-                elif threshold_flag < 0:
-                    threshold = -1 * threshold_flag * default_threshold
-                    log.info("Background threshold set to {} based on {}".format(threshold.max(),
-                                                                                 default_threshold.max()))
-                    bkg_rms_mean = threshold.max()
-                else:
-                    bkg_rms_mean = 3. * threshold_flag
-                    threshold = bkg_rms_mean
-
-                if bkg_rms_mean < 0:
-                    bkg_rms_mean = 0.
+                bkg_background_ra = bkg.background
+                bkg_rms_ra = bkg.background_rms
+                bkg_rms_median = bkg.background_rms_median
                 break
 
         # If Background2D does not work at all, define default scalar values for
         # the background to be used in source identification
         if bkg is None:
-            bkg_mean = bkg_rms_mean = max(0.01, self.data.min())
-            bkg_rms = nsigma * bkg_rms_mean
-            bkg_dao_rms = bkg_rms_mean
-            threshold = bkg_rms_mean + bkg_rms
-
-        # *** FIX: Need to do something for bkg if bkg is None ***
-
-        # Report other useful quantities
-        log.info("")
-        log.info("Mean background: {}".format(bkg_mean))
-        log.info("Mean threshold: {}".format(np.mean(threshold)))
-        log.info("")
-        log.info("{}".format("=" * 80))
+            log.info("Background2D failure detected. Using alternative background calculation instead....")
+            mask = make_source_mask(self.data, nsigma=2, npixels=5, dilate_size=11)
+            sigcl_mean, sigcl_median, sigcl_std = sigma_clipped_stats(self.data, sigma=3.0, mask=mask, maxiters=9)
+            bkg_rms_median = sigcl_std
+            # create background frame shaped like self.data populated with sigma-clipped median value
+            bkg_background_ra = np.full_like(self.data, sigcl_median)
+            # create background frame shaped like self.data populated with sigma-clipped standard deviation value
+            bkg_rms_ra = np.full_like(self.data, sigcl_std)
 
         self.bkg = bkg
-        self.bkg_dao_rms = bkg_dao_rms
-        self.bkg_rms_mean = bkg_rms_mean
-        self.threshold = threshold
+        self.bkg_background_ra = bkg_background_ra
```

Comment:
I would suggest that the outputs be the following: bkg_background_ra, bkg_median (derived from bkg.background_median or sigcl_median), bkg_rms_ra, and bkg_rms_median.  There seems to be no use for returning the Background2D object "bkg" directly since all our code needs are these specific values which (with this code) will always be generated/computed regardless of the success/failure of Background2D.

Reply:
Issue resolved in commit [c631ee9](https://github.com/spacetelescope/drizzlepac/pull/455/commits/c631ee978a71b8519922f8783c215cb65a6165a8)